lang/python/python-aiohttp-cors/Makefile

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright (C) 2019 CZ.NIC, z. s. p. o. (https://www.nic.cz/)
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=aiohttp-cors
+PKG_VERSION:=0.7.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=aiohttp-cors-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://files.pythonhosted.org/packages/source/a/aiohttp_cors/
+PKG_HASH:=4d39c6d7100fd9764ed1caf8cebf0eb01bf5e3f24e2e073fda6234bc48b19f5d
+
+PKG_MAINTAINER:=Josef Schlehofer <pepe.schlehofer@gmail.com>
+PKG_LICENSE:=Apache-2.0
+PKG_LICENSE_FILES:=LICENSE
+
+include $(INCLUDE_DIR)/package.mk
+include ../python3-package.mk
+
+define Package/python3-aiohttp-cors
+  SECTION:=lang
+  CATEGORY:=Languages
+  SUBMENU:=Python
+  TITLE:=CORS support for aiohttp
+  URL:=https://github.com/aio-libs/aiohttp-cors
+  DEPENDS:= \
+	+python3-light \
+	+python3-setuptools \
```

Comment:
From [setup.py](https://github.com/aio-libs/aiohttp-cors/blob/v0.7.0/setup.py#L55), it looks like this is only a setup requirement? If so, it should already be installed as part of host Python.

Reply:
To clarify, if this is only a setup requirement then it isn't necessary as an install dependency.

Reply:
Right. Setuptools was easier to write than separate dependencies like cgi, logging, multidict and so on. I will edit it.

Reply:
Added: `yarl`, `cgi`, `logging`, `multidict`, `async-timeout`

See log below:

```
root@turris:~# opkg install python3-aiohttp_3.5.4-1_aarch64_cortex-a53.ipk 
Installing python3-aiohttp (3.5.4-1) to root...
Installing python3-base (3.7.2-3.7-13.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-base_3.7.2-3.7-13_aarch64_cortex-a53.ipk
Installing python3-light (3.7.2-3.7-13.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-light_3.7.2-3.7-13_aarch64_cortex-a53.ipk
Configuring python3-base.
Configuring python3-light.
Configuring python3-aiohttp.
root@turris:~# python3
Python 3.7.2 (default, Jun 18 2019, 12:52:18) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import aiohttp_cors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/__init__.py", line 20, in <module>
  File "/__init__.py", line 5, in <module>
  File "/hdrs.py", line 6, in <module>
ImportError: cannot import name 'istr' from 'multidict' (unknown location)
>>> 
root@turris:~# opkg install python3-multidict
Installing python3-multidict (4.5.2-3.7-1.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-multidict_4.5.2-3.7-1_aarch64_cortex-a53.ipk
Installing python3-attrs (19.1.0-3.7-1.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-attrs_19.1.0-3.7-1_aarch64_cortex-a53.ipk
Configuring python3-attrs.
Configuring python3-multidict.
root@turris:~# python3
Python 3.7.2 (default, Jun 18 2019, 12:52:18) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import aiohttp_cors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/__init__.py", line 20, in <module>
  File "/__init__.py", line 6, in <module>
  File "/client.py", line 30, in <module>
ImportError: cannot import name 'URL' from 'yarl' (unknown location)
>>> 
root@turris:~# opkg install python3-yarl
Installing python3-yarl (1.3.0-3.7-1.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-yarl_1.3.0-3.7-1_aarch64_cortex-a53.ipk
Installing python3-email (3.7.2-3.7-13.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-email_3.7.2-3.7-13_aarch64_cortex-a53.ipk
Installing python3-urllib (3.7.2-3.7-13.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-urllib_3.7.2-3.7-13_aarch64_cortex-a53.ipk
Installing python3-codecs (3.7.2-3.7-13.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-codecs_3.7.2-3.7-13_aarch64_cortex-a53.ipk
Installing python3-idna (2.8-3.7-2.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-idna_2.8-3.7-2_aarch64_cortex-a53.ipk
Configuring python3-email.
Configuring python3-urllib.
Configuring python3-codecs.
Configuring python3-idna.
Configuring python3-yarl.
root@turris:~# python3
Python 3.7.2 (default, Jun 18 2019, 12:52:18) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import aiohttp_cors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/__init__.py", line 20, in <module>
  File "/__init__.py", line 6, in <module>
  File "/client.py", line 32, in <module>
  File "/http.py", line 7, in <module>
  File "/http_parser.py", line 14, in <module>
  File "/base_protocol.py", line 4, in <module>
  File "/tcp_helpers.py", line 20, in <module>
AttributeError: module 'asyncio' has no attribute 'Transport'
>>> 
root@turris:~# opkg list-installed | grep python3
python3-aiohttp - 3.5.4-1
python3-aiohttp-cors - 0.7.0-1
python3-attrs - 19.1.0-3.7-1.3
python3-base - 3.7.2-3.7-13.3
python3-codecs - 3.7.2-3.7-13.3
python3-email - 3.7.2-3.7-13.3
python3-idna - 2.8-3.7-2.3
python3-light - 3.7.2-3.7-13.3
python3-multidict - 4.5.2-3.7-1.3
python3-urllib - 3.7.2-3.7-13.3
python3-yarl - 1.3.0-3.7-1.3
root@turris:~# opkg install python3-asyncio
Installing python3-asyncio (3.7.2-3.7-13.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-asyncio_3.7.2-3.7-13_aarch64_cortex-a53.ipk
Configuring python3-asyncio.
root@turris:~# python3
Python 3.7.2 (default, Jun 18 2019, 12:52:18) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import aiohttp_cors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/__init__.py", line 20, in <module>
  File "/__init__.py", line 6, in <module>
  File "/client.py", line 3, in <module>
  File "/__init__.py", line 8, in <module>
  File "/base_events.py", line 18, in <module>
  File "/__init__.py", line 8, in <module>
  File "/_base.py", line 42, in <module>
AttributeError: module 'logging' has no attribute 'getLogger'
>>> 
root@turris:~# opkg install python3-logging
Installing python3-logging (3.7.2-3.7-13.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-logging_3.7.2-3.7-13_aarch64_cortex-a53.ipk
Configuring python3-logging.
root@turris:~# python3
Python 3.7.2 (default, Jun 18 2019, 12:52:18) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import aiohttp_cors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/__init__.py", line 20, in <module>
  File "/__init__.py", line 6, in <module>
  File "/client.py", line 32, in <module>
  File "/http.py", line 7, in <module>
  File "/http_parser.py", line 15, in <module>
  File "/helpers.py", line 6, in <module>
ModuleNotFoundError: No module named 'cgi'
>>> 
root@turris:~# opkg install python3-cgi
Installing python3-cgi (3.7.2-3.7-13.0) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-cgi_3.7.2-3.7-13_aarch64_cortex-a53.ipk
Configuring python3-cgi.
root@turris:~# python3
Python 3.7.2 (default, Jun 18 2019, 12:52:18) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import aiohttp_cors
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/__init__.py", line 20, in <module>
  File "/__init__.py", line 6, in <module>
  File "/client.py", line 32, in <module>
  File "/http.py", line 7, in <module>
  File "/http_parser.py", line 15, in <module>
  File "/helpers.py", line 596, in <module>
AttributeError: module 'async_timeout' has no attribute 'timeout'
>>>   
root@turris:~# opkg install python3-async-timeout
Installing python3-async-timeout (3.0.1-3.7-1.3) to root...
Downloading https://repo.turris.cz/master/mox/packages/packages/python3-async-timeout_3.0.1-3.7-1_aarch64_cortex-a53.ipk
Configuring python3-async-timeout.
root@turris:~# python3
Python 3.7.2 (default, Jun 18 2019, 12:52:18) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import aiohttp_cors

```

Reply:
Most of those packages are already stated in python3-aiohttp, but to be able to test it, which packages are necessary, I need to compile and install it without dependencies except python3-light as it is also dependency during compiling.

Reply:
I think all the errors you are seeing here was because aiohttp didn't have its dependencies, not because aiohttp-cors needed these packages.

aiohttp-cors has [only one explicit dependency](https://github.com/aio-libs/aiohttp-cors/blob/v0.7.0/setup.py#L68) (aiohttp). Looking at search results for the other packages ([asyncio](https://github.com/aio-libs/aiohttp-cors/search?q=asyncio), [async-timeout](https://github.com/aio-libs/aiohttp-cors/search?q=async_timeout), [cgi](https://github.com/aio-libs/aiohttp-cors/search?q=cgi), [logging](https://github.com/aio-libs/aiohttp-cors/search?q=logging), [multidict](https://github.com/aio-libs/aiohttp-cors/search?q=multidict), [yarl](https://github.com/aio-libs/aiohttp-cors/search?q=yarl)) appears to confirm my belief that aiohttp-cors does not directly use these packages.

(Conversely, looking at search results for aiohttp ([asyncio](https://github.com/aio-libs/aiohttp/search?q=asyncio), [async-timeout](https://github.com/aio-libs/aiohttp/search?q=async_timeout), [cgi](https://github.com/aio-libs/aiohttp/search?q=cgi), [logging](https://github.com/aio-libs/aiohttp/search?q=logging), [multidict](https://github.com/aio-libs/aiohttp/search?q=multidict), [yarl](https://github.com/aio-libs/aiohttp/search?q=yarl)) confirms that aiohttp does depend on these packages.)

I would prefer to have this package depend on its explicit dependencies only :laughing:

Reply:
Ok, removed all the dependencies, which I added yesterday as well with setuptools.